### PR TITLE
metrics-prometheus-collector: Add leases to ClusterRole

### PR DESCRIPTION
**Pull request by mrueg**
metrics-prometheus-collector requires access to leases API for leadership election.

```
E0126 13:26:43.567658       1 leaderelection.go:367] Failed to update lock: leases.coordination.k8s.io "capacity-metrics-prometheus-collector-lock" is forbidden: User "system:serviceaccount:usage-metrics-collector:metrics-prometheus-collector" cannot update resource "leases" in API group "coordination.k8s.io" in the namespace "usage-metrics-collector"
```

### CODE REVIEW

**Comment by ehashman** [comment]
@mrueg can you please PR yourself into kubernetes-sigs membership so we don't need to apply /ok-to-test to your PRs? https://github.com/kubernetes/org/blob/main/config/kubernetes-sigs/org.yaml#L16 As a Kubernetes org member, you get kubernetes-sigs membership as well.

**Comment by mrueg** [comment]
@ehashman sure, I created https://github.com/kubernetes/org/pull/3987 Thanks for letting me know! I wasn't aware. :+1:

**Comment by ehashman** [comment]
No worries, should make it a little easier going forward :) just gotta wait on a GH admin to approve.

File: config/metrics-prometheus-collector/collector.role.yaml [hunk 1]
```diff
@@ -6,3 +6,6 @@ rules:
 - apiGroups:  [ "", "apps", "batch", "quotamanagement.usagemetricscollector.sigs.k8s.io"]
   resources: [ "*" ]
   verbs: [ get, list, watch ]
+- apiGroups: ["coordination.k8s.io"]
+  resources: ["leases"]
+  verbs: [ "*" ]
```
Comment (ehashman) [inline config/metrics-prometheus-collector/collector.role.yaml:11]: Can we constrain the verbs more? This should be sufficient:

```
verbs:
  - get
  - create
  - patch
  - update
  - delete
```

Comment (mrueg) [reply config/metrics-prometheus-collector/collector.role.yaml]: Thanks, updated the verbs!

**Comment by mrueg** [comment]
I've also added RBAC to create events, since I saw that popping up in the logs as well:

```
E0126 18:37:41.706864       1 event.go:267] Server rejected event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"capacity-metrics-prometheus-collector-lock.
...
 'events is forbidden: User "system:serviceaccount:usage-metrics-collector:metrics-prometheus-collector" cannot create resource "events" in API group "" in the namespace "usage-metrics-collector"' (will not retry!)
```

File: config/metrics-prometheus-collector/collector.role.yaml [hunk 2]
```diff
@@ -6,3 +6,9 @@ rules:
 - apiGroups:  [ "", "apps", "batch", "quotamanagement.usagemetricscollector.sigs.k8s.io"]
   resources: [ "*" ]
   verbs: [ get, list, watch ]
+- apiGroups: ["coordination.k8s.io"]
+  resources: ["leases"]
+  verbs: [ get, create, patch, update, delete ]
+- apiGroups: [""]
+  resources: ["events"]
+  verbs: [ create ]
```
Comment (ehashman) [inline config/metrics-prometheus-collector/collector.role.yaml:14]: might also need watch

Comment (mrueg) [reply config/metrics-prometheus-collector/collector.role.yaml]: Thanks, added!

**Review by ehashman — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**